packages/compass-web/sandbox/index.tsx

```diff
@@ -0,0 +1,242 @@
+import React, { useCallback, useState } from 'react';
+import ReactDOM from 'react-dom';
+import {
+  TextArea,
+  Button,
+  resetGlobalCSS,
+  Card,
+  KeylineCard,
+  css,
+  spacing,
+  palette,
+  Label,
+  ErrorBoundary,
+  Banner,
+} from '@mongodb-js/compass-components';
+import {
+  redactConnectionString,
+  ConnectionString,
+} from 'mongodb-connection-string-url';
+import { CompassWeb } from '../src/index';
+
+const sandboxContainerStyles = css({
+  paddingTop: spacing[6],
+  width: '100%',
+  height: '100%',
+});
+
+const cardStyles = css({
+  width: '50%',
+  maxWidth: spacing[6] * 10,
+  minWidth: spacing[6] * 6,
+  marginLeft: 'auto',
+  marginRight: 'auto',
+});
+
+const connectionFormStyles = css({
+  display: 'grid',
+  gridTemplateColumns: '100%',
+  gridAutoRows: 'auto',
+  gap: spacing[3],
+});
+
+const historyListStyles = css({
+  all: 'unset',
+  marginTop: spacing[1],
+  display: 'grid',
+  gridTemplateColumns: '100%',
+  gridAutoRows: 'auto',
+  gap: spacing[2],
+});
+
+const historyListItemStyles = css({
+  listStyle: 'none',
+  paddingTop: spacing[2],
+  paddingBottom: spacing[2],
+  paddingLeft: spacing[2],
+  paddingRight: spacing[2],
+});
+
+const historyItemButtonStyles = css({
+  all: 'unset',
+  display: 'block',
+  width: '100%',
+  cursor: 'pointer',
+  color: palette.black,
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+});
+
+resetGlobalCSS();
+
+function getHistory(): string[] {
+  try {
+    const b64Str = localStorage.getItem('CONNECTIONS_HISTORY');
+    if (!b64Str) {
+      return [];
+    }
+    const binStr = window.atob(b64Str);
+    const bytes = Uint8Array.from(binStr, (v) => v.codePointAt(0) ?? 0);
+    const str = new TextDecoder().decode(bytes);
+    return JSON.parse(str);
+  } catch (err) {
+    return [];
+  }
+}
+
+function saveHistory(history: any) {
+  try {
+    const bytes = new TextEncoder().encode(JSON.stringify(history));
+    const binStr = String.fromCodePoint(...bytes);
+    const b64Str = window.btoa(binStr);
```

Comment:
Wanted some sort of history for local sandbox and didn't want to store creds completely raw, this seems good enough for local dev, but tell me if you have any concerns, I'll remove this